digicampipe/image/utils.py

```diff
@@ -0,0 +1,416 @@
+    return -get_image_hexagonalicity(image_crop * mask_hexa, rotations=rotations)
+
+
+def reciprocal_to_lattice_space(a1, a2, shape):
+    # solution to a1.dot(b1) = shape, a2.dot(b1) = 0, a1.dot(b2) = 0, a2.dot(b2) = shape,
+    b1 = np.array((1 / (a1[0] - a1[1] / a2[1] * a2[0]), 1 / (a1[1] - a1[0] / a2[0] * a2[1]))) * shape
+    b2 = np.array((1 / (a2[0] - a2[1] / a1[1] * a1[0]), 1 / (a2[1] - a2[0] / a1[0] * a1[1]))) * shape
+    return (b1, b2)
+
+
+def moments2D(inpData):  # from https://github.com/indiajoe/HandyTools4Astronomers
+    """ Returns the (amplitude, xcenter, ycenter, xsigma, ysigma, rot, bkg, e) estimated from moments in the 2d input array Data """
+    # Taking median of the 4 edges points as background
+    bkg = np.median(np.hstack((inpData[0, :], inpData[-1, :], inpData[:, 0], inpData[:, -1])))
+    Data=  np.ma.masked_less(inpData-bkg, 0)   #Removing the background for calculating moments of pure 2D gaussian
+    # We also masked any negative values before measuring moments
+    amplitude = Data.max()
+    total = float(Data.sum())
+    Xcoords, Ycoords = np.indices(Data.shape)
+    xcenter = (Xcoords*Data).sum()/total
+    ycenter = (Ycoords*Data).sum()/total
+    RowCut = Data[int(xcenter), :]  # Cut along the row of data near center of gaussian
+    ColumnCut = Data[:, int(ycenter)]  # Cut along the column of data near center of gaussian
+    xsigma = np.sqrt(np.ma.sum(ColumnCut * (np.arange(len(ColumnCut))-xcenter)**2)/ColumnCut.sum())
+    ysigma = np.sqrt(np.ma.sum(RowCut * (np.arange(len(RowCut))-ycenter)**2)/RowCut.sum())
+    #Ellipcity and position angle calculation
+    Mxx = np.ma.sum((Xcoords-xcenter)*(Xcoords-xcenter) * Data) / total
+    Myy = np.ma.sum((Ycoords-ycenter)*(Ycoords-ycenter) * Data) / total
+    Mxy = np.ma.sum((Xcoords-xcenter)*(Ycoords-ycenter) * Data) / total
+    e = np.sqrt((Mxx - Myy)**2 + (2*Mxy)**2) / (Mxx + Myy)
+    pa = 0.5 * np.arctan(2*Mxy / (Mxx - Myy))
+    rot = np.rad2deg(pa)
+    return amplitude, xcenter, ycenter, xsigma, ysigma, rot, bkg, e
+
+
+def Gaussian2D(amplitude, xcenter, ycenter, xsigma, ysigma, rot, bkg):  # from https://github.com/indiajoe/HandyTools4Astronomers
+    """ Returns a 2D Gaussian function with input parameters. rotation input rot should be in degress """
+    rot=np.deg2rad(rot)  # Converting to radians
+    Xc=xcenter*np.cos(rot) - ycenter*np.sin(rot)  # Centers in rotated coordinates
+    Yc=xcenter*np.sin(rot) + ycenter*np.cos(rot)
+    # Now lets define the 2D gaussian function
+
+    def Gauss2D(x, y):
+        """ Returns the values of the defined 2d gaussian at x,y """
+        xr = x * np.cos(rot) - y * np.sin(rot)  # X position in rotated coordinates
+        yr = x * np.sin(rot) + y * np.cos(rot)
+        return amplitude * np.exp(-(((xr - Xc) / xsigma) ** 2 +((yr - Yc) / ysigma) ** 2) / 2) + bkg
+    return Gauss2D
+
+
+def FitGauss2D(Data, ip=None):  # from https://github.com/indiajoe/HandyTools4Astronomers
```

Comment:
`Data`: should not be capitalized
`ip` is not a very good parameter name, `initial_parameters` should be fine, I guess.